### PR TITLE
Updated to work with ES5 on AWS

### PR DESCRIPTION
**Pull request by cliffom**
Elasticsearch 5 on AWS does not allow signed requests with query params and instead required signed authorization headers. This PR modifies the signing methods when Elasticsearch is being used as a AWS service.

### CODE REVIEW

File: lib/elastic/index.ex [hunk 1]
```diff
@@ -31,7 +31,7 @@ defmodule Elastic.Index do
   """
 
   def create(index) do
-    HTTP.post(name(index))
+    HTTP.put(name(index))
```
Comment (nicholasjhenry) [inline lib/elastic/index.ex:34]: Out of curiosity, why did this need to change?

**Review by nicholasjhenry — APPROVED** [review]
Looks good to me, @cliffom.

Question: should we merge this into a `hollar` branch? I've always found it helpful to keep `master` reflective of the upstream repo.

File: lib/elastic/index.ex (hunk 1 above)
Comment (cliffom) [reply lib/elastic/index.ex:34]: @nicholasjhenry Elasticsearch uses `PUT` for index creation:
[5.3 documentation](https://www.elastic.co/guide/en/elasticsearch/reference/5.3/indices-create-index.html), [2.3 documentation](https://www.elastic.co/guide/en/elasticsearch/reference/2.3/indices-create-index.html)

Using `POST` for index creation apparently works with 2.3 (in our setup) but does not work with 5.3.

**Comment by cliffom** [comment]
@nicholasjhenry I like that idea. I'll branch `hollar` from master and modify this PR to be against that branch. I'm going to notify [radar/elastic](https://github.com/radar/elastic) of the changes and see if we can get a PR merged upstream.

**Comment by nicholasjhenry** [comment]
> Elasticsearch 5 on AWS does not allow signed requests with query params and instead required signed authorization headers

@cliffom is this documented on AWS or something you discovered when trying to get this running? I would like to get us off the fork and try to get these changes merged upstream.

**Pull request merged**